bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -55,20 +52,23 @@
 
     static final Logger LOGGER = Loggers.getLogger("ObjectId");
 
+    private static final int OBJECT_ID_LENGTH = 12;
     private static final int LOW_ORDER_THREE_BYTES = 0x00ffffff;
 
-    private static final int MACHINE_IDENTIFIER;
-    private static final short PROCESS_IDENTIFIER;
+    // Use primitives to represent the 5-byte random value.
+    private static final int RANDOM_VALUE1;
+    private static final int RANDOM_VALUE2;
+
     private static final AtomicInteger NEXT_COUNTER = new AtomicInteger(new SecureRandom().nextInt());
 
     private static final char[] HEX_CHARS = new char[] {
-      '0', '1', '2', '3', '4', '5', '6', '7',
-      '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
+            '0', '1', '2', '3', '4', '5', '6', '7',
+            '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
 
     private final int timestamp;
-    private final int machineIdentifier;
-    private final short processIdentifier;
     private final int counter;
+    private final int randomValue1;
+    private final int randomValue2;
```

Comment:
Any reason it's not still a short?  We only use 1 byte of it right?

Reply:
I am using 3 bytes from `randomValue1` and 2 bytes from `randomValue2`. But it would make sense to make `randomValue2` a short.